Use correct graph and new tasks

benjay10 opened the pull request:
Some changes to the model of Jobs, Tasks and files came to be while working an the `automatic-submission-service` a few months ago, but this service was accidentally left out. This PR tries to update this service to accommodate those changes.

* The updated Jobs and Tasks model means that the correct Task has to identified by its operation.
* The Job, other Tasks for the same Job, and related DataContainers and HarvestingCollections are also removed.
* All data operations now only take place in the correct graph for that submission.

benjay10 committed:
Use new model for tasks and jobs

benjay10 committed:
Find submission graph and use in all queries

All data about tasks, jobs and the submission is stored in the graph of
the organisation. Use the organisation UUID to recreate the graph and
use that graph for all reads and deletes.

benjay10 committed:
Remove all data about tasks and job

Instead of just the one task for the automatic submission service.

benjay10 committed:
Also delete logical files and some model changes

benjay10 committed:
Remove unneeded dependency

cecemel commented on delete.js:
```diff
@@ -181,83 +231,139 @@ async function getHarvestedFiles(uri) {
+          dct:isPartOf ?job ;
+          ?tp ?to .
+
+        OPTIONAL {
```
Perhaps replacing the optional with a union, and the many option with values?
something along the lines

```
{
// the base query
}
UNION {
  VALUES ?predicate {
  task:resultsContainer
  ....
}
}
```

benjay10 replied:
That would be an interesting pattern. Let's see if that works!

benjay10 replied:
Turns out that is not a good solution, but the original solution was also very bad! Because of the `OPTIONAL` expressions, some variables might be unbound and following `OPTIONAL` patterns that use those variables can then bind to anything else that could be unrelated to the Task and Job from before. The solution is to put everything in a single `OPTIONAL` block and to repeat the `OPTIONAL` block with less specific patterns afterwards as a fallback. This new solution (that has been tested a bit better) is already committed.

cecemel commented on delete.js:
```diff
@@ -181,83 +231,139 @@ async function getHarvestedFiles(uri) {
+      }
+    } WHERE {
+      GRAPH ${sparqlEscapeUri(graph)} {
+        ${sparqlEscapeUri(taskUri)} dct:isPartOf ?job .
```
is `${sparqlEscapeUri(taskUri)}` the same as the `?task` in the latter part of the query? wouldn't be a bind better?

benjay10 replied:
No, because we are interested in *all* the tasks related to that job. We start from one of the tasks, the one that we have the URI of, ask for the parent job and then ask for all the tasks for that job.

benjay10 committed:
Condense query by using UNION and VALUES

Not much denser, but interesting pattern that allows for less OPTIONALS.

benjay10 committed:
NOPE: query was wrong and was also wrong before that

After some more testing, we need to rollback to the old query and even
add some to that in order not to delete random HarvestingCollections
that can bind to otherwise unbound variables.
